File: solutions/day1.py
def partOne(entries):
    for i in range(len(entries)):
        for j in range(i + 1, len(entries)):
            if entries[i] + entries[j] == 2020:
                return entries[i] * entries[j]


def partTwo(entries):
    for i in range(len(entries)):
        for j in range(i + 1, len(entries)):
            for k in range(i + 2, len(entries)):
                if entries[i] + entries[j] + entries[k] == 2020:
                    return entries[i] * entries[j] * entries[k]


def main():
    with open("01.txt") as f:
        content = list(map(lambda numStr: int(numStr), f.readlines()))
    print(partOne(content))
    print(partTwo(content))


if __name__ == "__main__":
    main()

File: 2019/day1.hs
module Day1 where

calculateFuel :: Int -> Int
calculateFuel m = (m `div` 3) - 2

partOne :: [Int] -> Int
partOne xs = sum (map calculateFuel xs)

fuelCounts :: Int -> [Int]
fuelCounts x | f < 0     = []
             | otherwise = f : fuelCounts f
  where f = calculateFuel x

partTwo :: [Int] -> Int
partTwo xs = sum (map (sum . fuelCounts) xs)

main :: IO ()
main = do
  content <- readFile "./inputs/day1.txt"
  let masses = map read (lines content) :: [Int]
  putStrLn ("Part 1: " ++ show (partOne masses))
  putStrLn ("Part 2: " ++ show (partTwo masses))
